examples/nlp/sentence_embeddings_with_sbert.py

```diff
@@ -0,0 +1,472 @@
+
+"""
+Let's try this example before training and compare it to the output after training.
+"""
+
+sentences = [
+    "today is a very sunny day",
+    "I am hungry, I will get my meal.",
+    "the dog is eating his food.",
+]
+query = ["the dog is enjoying his meal."]
+
+encoder = roberta_encoder
+
+sentence_embeddings = encoder(tf.constant(sentences))
+query_embedding = encoder(tf.constant(query))
+
+cosine_simalarities = tf.matmul(query_embedding, tf.transpose(sentence_embeddings))
+for i, sim in enumerate(cosine_simalarities[0]):
+    print(f"cosine similarity between sentence {i+1} and the query = {sim} ")
+
+"""
+For the training, we will use `MeanSquaredError()` loss function and `Adam()` optimizer
+with learning rate = 2e-5.
+"""
+
+roberta_siamese = regression_siamese(roberta_encoder)
+roberta_siamese.compile(
+    loss=keras.losses.MeanSquaredError(),
+    optimizer=keras.optimizers.Adam(2e-5),
+    metrics=[keras.metrics.MeanSquaredError()],
+)
+
+roberta_siamese.fit(stsb_train, validation_data=stsb_valid, epochs=1)
+
+"""
+Let's try the model after training, we will notice a huge difference in the output. That
+means that the model after fine-tuning is capable of producing semantically meaningful
+embeddings. where the semantically similar sentences have a small angle between them. and
+semantically dissimilar sentences have a large angle between them.
+"""
+
+sentences = [
+    "today is a very sunny day",
+    "I am hungry, I will get my meal.",
+    "the dog is eating his food.",
+]
+query = ["the dog is enjoying his food."]
+
+encoder = roberta_siamese.get_encoder()
+
+sentence_embeddings = encoder(tf.constant(sentences))
+query_embedding = encoder(tf.constant(query))
+
+cosine_simalarities = tf.matmul(query_embedding, tf.transpose(sentence_embeddings))
+for i, sim in enumerate(cosine_simalarities[0]):
+    print(f"cosine similarity between sentence {i+1} and the query = {sim} ")
+"""
+### Fine-tune Using Triplet Objective Function
+
+For the siamese network with the triplet objective function, three sentences are passed
+to the siamese network *anchor*, *positive*, and *negative* sentences. *anchor* and
+*positive* sentences are semantically similar, and *anchor* and *negative* sentences are
+semantically dissimilar. The objective is to minimize the distance between the *anchor*
+sentence and the *positive* sentence. and to maximize the distance between the *anchor*
+sentence and the *negative* sentence.
+"""
+
+"""
+#### Load the dataset
+
+We will use the wikipedia-sections-triplets dataset for fine-tuning. This data set
+consists of sentences derived from wikipedia website. It has a collection of 3 sentences
+*anchor*, *positive*, *negative*. *anchor* and *positive* are derived from the same
+section. *anchor* and *negative* are derived from different sections.
+
+This dataset has 1.8 million training triplets. and 220,000 test triplets. We will only
+use 1200 triplets for the training. and 300 for testing in this example.
```

Comment:
more sentences starting with lowercase and (I'll stop marking them now, but please scan through the rest of file.